f5router/f5router.go

```diff
@@ -482,17 +485,44 @@ func (r *F5Router) makeRouteRule(ru updateHTTP) (*bigipResources.Rule, error) {
 		Request: true,
 	}
 
+	uriString := ru.URI().String()
+
 	var c []*bigipResources.Condition
-	if true == strings.HasPrefix(ru.URI().String(), "*.") {
-		c = append(c, &bigipResources.Condition{
-			EndsWith: true,
-			Host:     true,
-			HTTPHost: true,
-			Name:     "0",
-			Index:    0,
-			Request:  true,
-			Values:   []string{strings.TrimPrefix(u.Host, "*")},
-		})
+	if strings.Count(uriString, "*") == 1 {
+		splits := strings.Split(u.Host, "*")
+		numSplits := len(splits)
+		ruleIndex := 0
+		if strings.HasPrefix(u.Host, splits[0]) {
+			if splits[0] != "" {
+				c = append(c, &bigipResources.Condition{
+					BeginsWith: true,
+					Host:       true,
+					HTTPHost:   true,
+					Name:       strconv.Itoa(ruleIndex),
+					Index:      ruleIndex,
+					Request:    true,
+					Values:     []string{splits[0]},
+				})
+				ruleIndex++
+			}
+			splits = splits[1:]
+			numSplits--
+		}
+		if strings.HasSuffix(u.Host, splits[numSplits-1]) {
+			if splits[numSplits-1] != "" {
+				c = append(c, &bigipResources.Condition{
+					EndsWith: true,
+					Host:     true,
+					HTTPHost: true,
+					Name:     strconv.Itoa(ruleIndex),
+					Index:    ruleIndex,
+					Request:  true,
+					Values:   []string{splits[numSplits-1]},
+				})
+			}
+		}
+	} else if strings.Count(uriString, "*") > 1 {
+		r.logger.Warn("f5-router-multiple-uri-wildcards", zap.String("uri", uriString))
```

Comment:
return nil and error if this case is met. That way rule can be more specific.

Reply:
Prevent multi ** from even reaching this level. Add checks in higher process* functions.

Reply:
Fixed.